fdbclient/FileBackupAgent.actor.cpp

```diff
@@ -4121,9 +4125,13 @@ struct StartFullRestoreTaskFunc : RestoreTaskFuncBase {
 		// Order does not matter, they will be put in order when written to the restoreFileMap below.
 		state std::vector<RestoreConfig::RestoreFile> files;
 
+		Version firstConsistentVersion = std::numeric_limits<int64_t>::min();
 		for (const RangeFile& f : restorable.get().ranges) {
 			files.push_back({ f.version, f.fileName, true, f.blockSize, f.fileSize });
+			firstConsistentVersion = std::max(firstConsistentVersion, f.version);
 		}
+		restore.firstConsistentVersion().set(tr, firstConsistentVersion);
+		wait(tr->commit());
```

Comment:
This commit must have its own retry loop.  The `getRestoreSet()` operation above can take several minutes or more, so a spurious failure here, which will cause the entire task to be run again, is very costly.

Great idea to commit this first, so a user watching restore status will see it during the queue load which can also take a few minutes.

Reply:
Yeah I added when resolving Jingyu's comment, but messed up the commit history so it's gone. Added again.

Reply:
@RenxuanW You resolved this comment but I do not see a change?  Still no retry loop.

Reply:
Yeah the branch was left in a weird state. Should be fine now.